src/csharp-stellar-base/ChangeTrustOperation.cs

```diff
@@ -55,12 +55,16 @@ public Builder(Generated.ChangeTrustOp op)
             public Builder(Asset asset, long limit)
             {
                 Asset = CheckNotNull(asset, "asset cannot be null.");
+                if(limit < 0)
+                {
+                    throw new ArgumentException("limit must be non-negative.");
+                }
                 Limit = CheckNotNull(limit, "limit cannot be null.");
             }
 
             public Builder SetSourceAccount(KeyPair sourceAccount)
             {
-                SourceAccount = CheckNotNull(sourceAccount, "sourceAccount cannot be null.");
+                SourceAccount = sourceAccount;
```

Comment:
why is this check gone? If you don't want the sourceaccount to be set, you just don't call this method

Reply:
I had in mind a case where the SourceAccount was not null, and you want it to be set to null.

Reply:
I would prefer a "RemoveSourceAccount" method for that case, but I think that will be rarely used. So add it when it is missed by someone